hoomd/md/pytest/test_pair.py

```diff
@@ -0,0 +1,751 @@
+                                        dict(zip(combos,
+                                                 dpdlj_valid_param_dicts)),
+                                        {"kT": 1}))
+
+    dlvo_arg_dict = {'kappa': [1.0, 2.0, 5.0],
+                     'Z': [0.1, 0.5, 2.0],
+                     'A': [0.1, 0.5, 2.0]}
+    dlvo_valid_param_dicts = _make_valid_param_dicts(dlvo_arg_dict)
+    valid_params_list.append(paramtuple(hoomd.md.pair.DLVO,
+                                        dict(zip(combos,
+                                                 dlvo_valid_param_dicts)),
+                                        {}))
+    return valid_params_list
+
+
+@pytest.fixture(scope="function", params=_valid_params(), ids=(lambda x: x[0].__name__))
+def valid_params(request):
+    return deepcopy(request.param)
+
+def test_valid_params(valid_params):
+    pot = valid_params.pair_potential(**valid_params.extra_args,
+                                      nlist=hoomd.md.nlist.Cell(),
+                                      r_cut=2.5,
+                                      mode='none')
+    for pair in valid_params.pair_potential_params:
+        pot.params[pair] = valid_params.pair_potential_params[pair]
+    _assert_equivalent_type_params(valid_params.pair_potential_params,
+                                   pot.params.to_dict())
+
+
+def _update_snap(pair_potential, snap):
+    if 'Ewald' in str(pair_potential) and snap.exists:
+        snap.particles.charge[:] = 1
+    elif 'SLJ' in str(pair_potential) and snap.exists:
+        snap.particles.diameter[:] = 2
+    elif 'DLVO' in str(pair_potential) and snap.exists:
+        snap.particles.diameter[0] = 0.2
+        snap.particles.diameter[1] = 0.5
+
+
+def test_attached_params(simulation_factory, lattice_snapshot_factory,
+                         valid_params):
+    pair_potential, pair_potential_dict, extra_args = valid_params
+    pair_keys = valid_params.pair_potential_params.keys()
+    particle_types = list(set(itertools.chain.from_iterable(pair_keys)))
+    pot = valid_params.pair_potential(**valid_params.extra_args,
+                                      nlist=hoomd.md.nlist.Cell(),
+                                      r_cut=2.5, mode='none')
+    for pair in valid_params.pair_potential_params:
+        pot.params[pair] = valid_params.pair_potential_params[pair]
+
+    snap = lattice_snapshot_factory(particle_types=particle_types,
+                                    n=10, a=1.5, r=0.01)
+
+    _update_snap(valid_params.pair_potential, snap)
+    if snap.exists:
+        snap.particles.typeid[:] = np.random.randint(0,
+                                                     len(snap.particles.types),
+                                                     snap.particles.N)
+    sim = simulation_factory(snap)
+    sim.operations.integrator = hoomd.md.Integrator(dt=0.005)
+    sim.operations.integrator.forces.append(pot)
+    sim.run(10)
```

Comment:
```suggestion
    sim.run(1)
```
We still run but this should test a bit faster.